GitCommands/Git/GitCreateTagCmd.cs

```diff
@@ -0,0 +1,145 @@
+        public string TagMessage = "";
+        public string SignKeyId = "";
+    }
+
+    public sealed class GitCreateTagCmd : GitCommand
+    {
+        public GitCreateTagArgs Args;
+        public string TagMessageFileName;
+
+        public GitCreateTagCmd(GitCreateTagArgs aArgs)
+        {
+            Args = aArgs;
+        }
+
+        public override string GitComandName()
+        {
+            return "tag";
+        }
+
+        public override IEnumerable<string> CollectArguments()
+        {
+            if (Args.Force)
+            {
+                yield return "-f";
+            }
+
+            string operationArg;
+            if (GetArgumentForOperation(out operationArg))
+            {
+                yield return operationArg;
+            }
+
+            if (Args.OperationType.CanProvideMessage())
+            {
+                yield return "-F " + TagMessageFileName.Quote();
+            }
+
+            yield return Args.TagName.Trim().Quote();
+            yield return "--";
+            yield return Args.Revision.Quote();
+        }
+
+        private bool GetArgumentForOperation(out string operationArg)
+        {
+            switch (Args.OperationType)
+            {
+                /* Lightweight */
+                case TagOperation.Lightweight:
+                    operationArg = null;
+                    return false;
+
+                /* Annotate */
+                case TagOperation.Annotate:
+                    operationArg = "-a";
+                    return true;
+
+                /* Sign with default GPG */
+                case TagOperation.SignWithDefaultKey:
+                    operationArg = "-s";
+                    return true;
+
+                /* Sign with specific GPG */
+                case TagOperation.SignWithSpecificKey:
+                    if (string.IsNullOrEmpty(Args.SignKeyId))
+                    {
+                        throw new ArgumentNullException("Args.SignKeyId");
+                    }
+                    operationArg = $"-u {Args.SignKeyId}";
+                    return true;
+
+                /* Error */
+                default:
+                    throw new NotSupportedException("Invalid TagOperation: " + Args.OperationType);
+            }
+        }
+
+        public override bool AccessesRemote()
+        {
+            return false;
+        }
+
+        public override bool ChangesRepoState()
+        {
+            return true;
+        }
+
+        public override void Validate()
+        {
+            if (string.IsNullOrEmpty(Args.Revision))
+            {
+                throw new ArgumentNullException("Args.Revision");
+            }
+
+            if (string.IsNullOrEmpty(Args.TagName))
+            {
+                throw new ArgumentNullException("Args.TagName");
+            }
+
+            if (Args.OperationType.CanProvideMessage() && TagMessageFileName.IsNullOrEmpty())
```

Comment:
Because `string.IsNullOrEmpty()` was copied from the prev implementation. I added `TagMessageFileName` and prefer to use extension method for that - it reads better for me.
But it is only my opinion, that is why I did not change others code to meet my personal preferences.
I see no reason why we have to waste time for things like that. It is semantically equal. Does it bring any danger to the code base?

Reply:
No danger, but inconsistency. Less consistency = more entropy = more maintainability overhead

Reply:
When I make a change I like to be focused on that change. I don't like correcting other devs preferences and mixing it to the PR - it carry a risk of introducing a bug while performing such corrections. 
TagMessageFileName.IsNullOrEmpty is closer to the natural human language than string.IsNullOrEmpty(TagMessageFileName)
If that inconsistency bothers you we can add a rule to our coding guidelines to make it clear which form should be used and then we can validate new PRs against that rule.